events/signals.py

```diff
@@ -0,0 +1,27 @@
+import logging
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from events.models import Event
+from notifications.enums import NotificationType
+from notifications.utils import send_notification
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=Event)
+def handle_event_save(sender, instance, created, **kwargs):
+    if not created:
+        return
+
+    email = instance.contract_zone.get_contact_email()
+    if not email:
+        logger.warning(
+            'Contract zone {} has no contact email so cannot send "event created" notification.'.format(
+                instance.contract_zone
+            )
+        )
+        return
+
+    send_notification(email, NotificationType.EVENT_CREATED, {'event': instance})
```

Comment:
why are you not using `get_notification_context`?

Reply:
I could have (and initially did), but figured this way it will be more flexible when the event's all data is available, as we don't know what information will be needed in the notification.

That said, one potential problem I can see with this is that it might be dangerous to have the actual object available in the mail template. The template is rendered in Jinja sandbox and all, but still, dunno. This was the original reason I started using `get_notification_context()` which returns just strings in the first place :)

I was thinking this project could be a kind of guinea pig for trying out how well just using the whole object in the template will work.